src/gui/qgsexpressionbuilderwidget.h

```diff
@@ -547,47 +428,26 @@ class GUI_EXPORT QgsExpressionBuilderWidget : public QWidget, private Ui::QgsExp
      */
     void setParserError( bool parserError );
 
-    void loadFieldValues( const QVariantMap &values );
-
-    void loadFieldsAndValues( const QMap<QString, QVariantMap> &fieldValues );
-
-    /**
-     * Display a message box to ask the user what to do when an expression
-     * with the same \a label already exists. Answering "Yes" will replace
-     * the old expression with the one from the file, while "No" will keep
-     * the old expression.
-     * \param isApplyToAll whether the decision of the user should be applied to any future label collision
-     * \param isOkToOverwrite whether to overwrite the old expression with the new one in case of label collision
-     * \param label the label of the expression
-     * \param oldExpression the old expression for a given label
-     * \param newExpression the new expression for a given label
-     * \since QGIS 3.14
-     */
-    void showMessageBoxConfirmExpressionOverwrite( bool &isApplyToAll, bool &isOkToOverwrite, const QString &label, const QString &oldExpression, const QString &newExpression );
-
-    bool mAutoSave = true;
-    QString mFunctionsPath;
-    QgsVectorLayer *mLayer = nullptr;
-    std::unique_ptr<QStandardItemModel> mModel;
     // Will hold items with
     // * a display string that matches the represented field values
     // * custom data in Qt::UserRole + 1 that contains a ready to use expression literal ('quoted string' or NULL or a plain number )
     std::unique_ptr<QStandardItemModel> mValuesModel;
     std::unique_ptr<QSortFilterProxyModel> mProxyValues;
-    std::unique_ptr<QgsExpressionItemSearchProxy> mProxyModel;
-    QMap<QString, QgsExpressionItem *> mExpressionGroups;
+
+    ExpressionTreeMenuProvider *mExpressionTreeMenuProvider = nullptr;
```

Comment:
Unique ptr candidate

Reply:
as discussed I had troubles with this one.